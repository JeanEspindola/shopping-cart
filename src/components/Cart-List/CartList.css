.cartlist__table__header {
  background-color: #25258A;
  color: white;
}

.cartlist__table__body {
  background-color: white;
  color: gray;
}