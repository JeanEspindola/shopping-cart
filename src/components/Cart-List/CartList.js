/*
* Cart List Presentation Component.
*
* Renders the list and passes values to CartItem, child component.
*
*/
import React from 'react';
import { Row, Col, Table } from 'react-bootstrap';
import PropTypes from 'prop-types';
import CartItem from '../Cart-Item/CartitemContainer';
import './CartList.css';

const CartList = ({ list }) => {
  if (list.length === 0) {
    return (
      <p>There are no items in your cart.</p>
    );
  }

  return (
    <Row className="cartlist">
      <Col xs={12}>
        <Table className="cartlist__table">
          <thead className="cartlist__table__header">
            <tr>
              <th>#</th>
              <th>Name</th>
              <th>Comments</th>
              <th>Price</th>
              <th>Tax</th>
              <th className="cartlist__table__header__total">Total Item</th>
              <th className="cartlist__table__header__actions" />
            </tr>
          </thead>
          <tbody className="cartlist__table__body">
            {list.map(item =>
              (<CartItem
                key={item.id}
                id={item.id}
                product={item.product}
                comments={item.comments}
                price={item.price}
                tax={item.tax}
                total={item.totalItem}
              />))
            }
          </tbody>
        </Table>
      </Col>
    </Row>
  );
};

CartList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default CartList;
