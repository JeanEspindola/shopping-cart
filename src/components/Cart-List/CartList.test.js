import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CartList from './CartList';

describe('CartList', () => {
  let list;
  let Component;

  it('renders CartList without crashing', () => {
    list = [{ id: 1 }, { id: 2 }];
    Component = shallow(<CartList list={list} />);

    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });

  it('renders CartList with list props empty', () => {
    const text = 'There are no items in your cart.';
    list = [];
    Component = shallow(<CartList list={list} />);

    expect(Component.find('p').text()).toBe(text);
  });
});
