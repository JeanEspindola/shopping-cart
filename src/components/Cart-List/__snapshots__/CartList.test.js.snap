// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartList renders CartList without crashing 1`] = `
<Row
  bsClass="row"
  className="cartlist"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <Table
      bordered={false}
      bsClass="table"
      className="cartlist__table"
      condensed={false}
      hover={false}
      responsive={false}
      striped={false}
    >
      <thead
        className="cartlist__table__header"
      >
        <tr>
          <th>
            #
          </th>
          <th>
            Name
          </th>
          <th>
            Comments
          </th>
          <th>
            Price
          </th>
          <th>
            Tax
          </th>
          <th
            className="cartlist__table__header__total"
          >
            Total Item
          </th>
          <th
            className="cartlist__table__header__actions"
          />
        </tr>
      </thead>
      <tbody
        className="cartlist__table__body"
      >
        <Connect(CartItem)
          id={1}
          key="1"
        />
        <Connect(CartItem)
          id={2}
          key="2"
        />
      </tbody>
    </Table>
  </Col>
</Row>
`;
